src/libs/Middleware/Reauthentication.js

```diff
@@ -0,0 +1,95 @@
+import lodashGet from 'lodash/get';
+import CONST from '../../CONST';
+import * as NetworkStore from '../Network/NetworkStore';
+import * as MainQueue from '../Network/MainQueue';
+import * as Authentication from '../Authentication';
+import * as PersistedRequests from '../actions/PersistedRequests';
+import * as Request from '../Request';
+import Log from '../Log';
+
+/**
+ * Reauthentication middleware
+ *
+ * @param {Promise} response
+ * @param {Object} request
+ * @param {Boolean} isFromSequentialQueue
+ * @returns {Promise}
+ */
+function Reauthentication(response, request, isFromSequentialQueue) {
+    return response
+        .then((data) => {
+            if (NetworkStore.isOffline()) {
+                // If we are offline and somehow handling this response we do not want to reauthenticate
+                throw new Error('Unable to reauthenticate because we are offline');
+            }
```

Comment:
This PR causes a regression [here](https://github.com/Expensify/App/issues/16637) more details on the [PR](https://github.com/Expensify/App/pull/18237/files)